### ISSUE TITLE
[BUG] For block ciphers with block size <16, CTR mode doesn't mutate the counter

### ISSUE THREAD

**Issue by cbeuw** [digitalbazaar/forge#721]
For instance
```js
var forge = require('node-forge');

var key = forge.util.hexToBytes('0011223344556677')
var input = "The quick brown fox jumps over the lazy dog"
var iv = forge.util.hexToBytes('00')

var cipher = forge.cipher.createCipher('DES-CTR', key);
cipher.start({iv:iv});
cipher.update(forge.util.createBuffer(input));
cipher.finish();
console.log(cipher.output.toHex())
```
Yields `700abe5fad7509b94f42b90db3770efa420da35fb6750daa5742b409b97240ae4c07fb13bd7a19fa400dbc51`
However, both GoLang's standard library and pyCryptoDome gives `700abe5fad7509b9bd9e9a3514e4a6d9536d0c9b2e51e7a2a611db37b45dd99a0c2064464a77a377f791b11b`

The results deviate after `700abe5fad7509b9`, which is exactly one block
```
wrong: 700abe5fad7509b94f42b90db3770efa420da35fb6750daa5742b409b97240ae4c07fb13bd7a19fa400dbc51
right: 700abe5fad7509b9bd9e9a3514e4a6d9536d0c9b2e51e7a2a611db37b45dd99a0c2064464a77a377f791b11b
```

This is due to that:
- In cipherModes.js, IV is transformed into an array of four 32-bit ints (which is 16 bytes in total) regardless of the blocksize 
https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/cipherModes.js#L447 https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/cipherModes.js#L972-L973 `_inBlock` is then set to this transformed IV to be fed into the algorithm's block encrypt function.

  When IV is 8 bytes in size, as is the case for ciphers with block size of 8, the first two elements of the initial `_inBlock` will be the correct two int32s representing the entire IV, and the last two will be 0

- In des.js, only the first two int32 of the input block are taken into account. https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/des.js#L364-L365 Since the first two elements do indeed represent the IV, the result of the first block is correct

- After one block has been encrypted, the **last element of _inBlock is incremented by 1** https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/cipherModes.js#L503-L504 https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/cipherModes.js#L979-L982 Recall the previous point that only the **first two elements of _inBlock** are accessed by the block encryption function, the next block will have the same input as the previous one. **The counter effectively hasn't been mutated at all**.

### Consequences
This bug means that all plaintext blocks are XORed with the same thing to produce the ciphertext. Therefore `Cipherblock1 ^ Cipherblock2` will result in `Plainblock1 ^ Plainblock2`. Similarly, `Ciberblock2 ^ Cipherblock3 = Plainblock2 ^ Plainblock3`, and so on.  With one known plaintext block at any position (it's also very easy to brute force), the entire ciphertext can be cascadingly decrypted.

### Why are other modes not affected?
Because in all other modes, the outputs are all "chained" into the input of the next block operation either directly, or after XORing. For block ciphers blocksize of 8, their block operations only alter the first two elements of the output block. This will alter the corresponding first two elements of the next input block, correctly chaining the operations and thus circumventing the bug
https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/des.js#L445-L446 And forge handles this correctly, taking blocksize into consideration. https://github.com/digitalbazaar/forge/blob/1af95997c4c555cf96e954a395c4295759f00fd7/lib/cipherModes.js#L236-L238

**Comment by davidlehn** [comment]
I merged the code fixes from #722 and the tests from #723 into master and released 0.9.1.  Hopefully this should catch IV issues in the future.  Please check this addresses your use case and close this issue.  Thanks.

**Comment by cbeuw** [comment]
Thanks!
